service/history/api/workflow_id_dedup.go

```diff
@@ -100,6 +104,39 @@ func ResolveDuplicateWorkflowID(
 	return nil, nil
 }
 
+func resolveJustStartedDuplicatedWorkflows(
+	shardContext shard.Context,
+	currentWorkflowStartTime time.Time,
+	newRunID string,
+	workflowID string,
+) (UpdateWorkflowActionFunc, error) {
+	// we are going to use "grace period"
+	// to prevent multiple calls to start worklfows with the same ID
+	// if new workflow is starting earlier then that period - we will not terminate old worklfow,
+	// but rather didn't start the new one
```

Comment:
Could this use full sentences? It's hard for me to tell where one thought ends and another begins.

PS: typo in `worklfows`
PPS: `rather didn't start the new one` should be `rather don't start the new one`

Reply:
change wording in the next rev.